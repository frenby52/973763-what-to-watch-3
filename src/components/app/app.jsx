import React from "react";
import PropTypes from "prop-types";
import {connect} from "react-redux";
import {BrowserRouter, Route, Switch} from "react-router-dom";
import Main from "../main/main.jsx";
import MoviePage from "../movie-page/movie-page.jsx";
import withActiveItem from '../../hocs/with-active-item/with-active-item';
import {ActionCreator} from "../../reducer/films/films";
const MoviePageWrapped = withActiveItem(MoviePage);
import {getFilteredMovieCards, getSelectedMovieId, isFullPlayerVisible} from "../../reducer/films/selectors.js";

const SIMILAR_FILMS_COUNT = 4;

class App extends React.PureComponent {
  constructor(props) {
    super(props);
  }

  _renderApp() {
    const {films, onCardClick, selectedMovieId, isFullVideoPlayerVisible, onVisibilityChange} = this.props;

    if (selectedMovieId >= 0) {
      const selectedMovie = films.find((film) => film.id === selectedMovieId);
      const similarFilms = films.filter((film) => film.id !== selectedMovieId && film.genre === selectedMovie.genre);

      return <MoviePageWrapped film={selectedMovie} similarFilms={similarFilms} onCardClick={onCardClick} isFullVideoPlayerVisible={isFullVideoPlayerVisible} onVisibilityChange={onVisibilityChange} />;
    }

    return <Main films={films} onCardClick={onCardClick} isFullVideoPlayerVisible={isFullVideoPlayerVisible} onVisibilityChange={onVisibilityChange} />;
  }

  render() {
    const {films, onCardClick, isFullVideoPlayerVisible, onVisibilityChange} = this.props;

    return (
      <BrowserRouter>
        <Switch>
          <Route exact path="/">
            {this._renderApp()}
          </Route>
          <Route exact path="/dev-film">
            <MoviePageWrapped film={films[0]} similarFilms={films.slice(0, SIMILAR_FILMS_COUNT)} onCardClick={onCardClick} isFullVideoPlayerVisible={isFullVideoPlayerVisible} onVisibilityChange={onVisibilityChange} />
          </Route>
        </Switch>
      </BrowserRouter>
    );
  }
}

App.propTypes = {
  onCardClick: PropTypes.func.isRequired,
  onVisibilityChange: PropTypes.func.isRequired,
  selectedMovieId: PropTypes.number.isRequired,
  isFullVideoPlayerVisible: PropTypes.bool.isRequired,
  films: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.number.isRequired,
    title: PropTypes.string.isRequired,
    previewImage: PropTypes.string.isRequired,
    genre: PropTypes.string.isRequired,
    releaseDate: PropTypes.number.isRequired,
    posterImage: PropTypes.string.isRequired,
    backgroundImage: PropTypes.string.isRequired,
    ratingScore: PropTypes.number.isRequired,
    ratingCount: PropTypes.number.isRequired,
    description: PropTypes.string.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    previewSrc: PropTypes.string.isRequired,
    runTime: PropTypes.number.isRequired,
  })).isRequired
};


const mapStateToProps = (state) => ({
  films: getFilteredMovieCards(state),
  selectedMovieId: getSelectedMovieId(state),
  isFullVideoPlayerVisible: isFullPlayerVisible(state)
});

const mapDispatchToProps = (dispatch) => ({
  onCardClick(id) {
    dispatch(ActionCreator.setMovieCardId(id));
  },
  onVisibilityChange() {
    dispatch(ActionCreator.changeVisibility());
  }
});

export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);
