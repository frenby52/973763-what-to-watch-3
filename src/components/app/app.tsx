import * as React from "react";
import {connect} from "react-redux";
import {Router, Route, Switch} from "react-router-dom";
import Main from "../main/main";
import MoviePage from "../movie-page/movie-page";
import SignIn from "../sign-in/sign-in";
import AddReview from "../add-review/add-review";
import MyList from "../my-list/my-list";
import withIsValid from '../../hocs/with-is-valid/with-is-valid';
import withActiveItem from '../../hocs/with-active-item/with-active-item';
import {getFilteredMovieCards, isAppLoading, getPromoFilm} from "../../reducer/films/selectors";
import Loader from "../loader/loader";
import {Operation as UserOperation} from "../../reducer/user/user";
import {Operation as CommentsOperation} from "../../reducer/comments/comments";
import history from "../../history";
import PrivateRoute from "../private-route/private-route";
import FullVideoPlayer from "../full-video-player/full-video-player";
import withFullPlayer from "../../hocs/with-full-player/with-full-player";
import {getMovieById, getSimilarFilms} from "../../utils";
import {isAuth} from "../../reducer/user/selectors";
import {Film} from "../../types";

const FullVideoPlayerWrapped = withFullPlayer(FullVideoPlayer);
const MoviePageWrapped = withActiveItem(MoviePage);
const AddReviewWrapped = withIsValid(AddReview);

type AppProps = {
  login: ({email, password}: {email: string; password: string}) => void;
  isAuthed: boolean;
  isLoading: boolean;
  films: Film[];
  promoFilm: Film;
  onCardClick: (id: number | string) => void;
};

const App: React.FunctionComponent<AppProps> = (props: AppProps) => {
  const {films, onCardClick, login, isLoading, promoFilm, isAuthed} = props;

  return (
    <Router history={history}>
      <Switch>
        <Route exact path={`/`}
          render={() => isLoading ? <Loader/> : <Main films={films} promoFilm={promoFilm} onCardClick={onCardClick} />}
        />
        <Route exact path={`/films/:id`}
          render={(routerProps) => isLoading ? <Loader/> : <MoviePageWrapped film={getMovieById(routerProps, films)} similarFilms={getSimilarFilms(routerProps, films)} onCardClick={onCardClick} />}
        />
        <Route exact path={`/login`}
          render={() => isAuthed ? history.push(`/`) : <SignIn onFormSubmit={login} />}
        />
        <Route
          exact
          path={`/player/:id`}
          render={(routerProps) => isLoading ? <Loader/> : <FullVideoPlayerWrapped onExitButtonClick = {routerProps.history.goBack} film = {getMovieById(routerProps, films) || promoFilm} autoPlay = {true} muted = {false}/>}
        />
        <PrivateRoute
          exact
          path={`/films/:id/review`}
          render={(routerProps) => <AddReviewWrapped film={getMovieById(routerProps, films)} />}
        />
        <PrivateRoute
          exact
          path={`/mylist`}
          render={() => (
            <MyList onCardClick={onCardClick} />
          )}
        />
      </Switch>
    </Router>
  );
};

const mapStateToProps = (state) => ({
  films: getFilteredMovieCards(state),
  promoFilm: getPromoFilm(state),
  isLoading: isAppLoading(state),
  isAuthed: isAuth(state),
});

const mapDispatchToProps = (dispatch) => ({
  onCardClick(id) {
    dispatch(CommentsOperation.getComments(id));
  },
  login(data) {
    dispatch(UserOperation.login(data));
  }
});

export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);
