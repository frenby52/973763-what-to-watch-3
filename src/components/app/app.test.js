import React from "react";
import renderer from "react-test-renderer";
import {App} from "./app.jsx";
import configureStore from "redux-mock-store";
import Namespace from "../../reducer/name-space.js";
import {Provider} from "react-redux";

const films = [
  {
    id: 0,
    title: `The Grand Budapest Hotel`,
    previewImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Drama`,
    releaseDate: 2014,
    posterImage: `the-grand-budapest-hotel-poster.jpg`,
    backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8.9,
    ratingCount: 240,
    description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
    director: `Wes Andreson`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 20,
  },
  {
    id: 1,
    title: `Bohemian Rhapsody`,
    previewImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Comedy`,
    releaseDate: 2011,
    posterImage: `the-grand-budapest-hotel-poster.jpg`,
    backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8,
    ratingCount: 1240,
    description: `Lorem ipsum dolor sit amet, consectetur adipiscing elit.`,
    director: `director2`,
    starring: [`actor`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 30,
  },
  {
    id: 2,
    title: `Macbeth`,
    previewImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Horror`,
    releaseDate: 2010,
    posterImage: `the-grand-budapest-hotel-poster.jpg`,
    backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 7,
    ratingCount: 40,
    description: `Cras aliquet varius magna, non porta ligula feugiat eget.`,
    director: `director3`,
    starring: [`Bill Murray`, `actor`, `Jude Law`, `Willem Dafoe`],
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 50,
  },
  {
    id: 3,
    title: `Aviator`,
    previewImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Comedy`,
    releaseDate: 2012,
    posterImage: `the-grand-budapest-hotel-poster.jpg`,
    backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 1.9,
    ratingCount: 2240,
    description: `Fusce tristique felis at fermentum pharetra.`,
    director: `director4`,
    starring: [`Bill Murray`, `Edward Norton`, `actor`, `Willem Dafoe`],
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 40,
  },
  {
    id: 4,
    title: `We need to talk about Kevin`,
    previewImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Crime`,
    releaseDate: 2013,
    posterImage: `the-grand-budapest-hotel-poster.jpg`,
    backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 2.9,
    ratingCount: 20,
    description: `Aliquam id orci ut lectus varius viverra.`,
    director: `director5`,
    starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `actor`],
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 20,
  },
  {
    id: 5,
    title: `What We Do in the Shadows`,
    previewImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Drama`,
    releaseDate: 2000,
    posterImage: `the-grand-budapest-hotel-poster.jpg`,
    backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 5.9,
    ratingCount: 240,
    description: `Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.`,
    director: `director6`,
    starring: [`actor`, `actor2`, `Jude Law`, `Willem Dafoe`],
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 31,
  },
  {
    id: 6,
    title: `Revenant`,
    previewImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Comedy`,
    releaseDate: 2001,
    posterImage: `the-grand-budapest-hotel-poster.jpg`,
    backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 4.9,
    ratingCount: 245,
    description: `Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.`,
    director: `director7`,
    starring: [`Bill Murray`, `actor`, `actor2`, `Willem Dafoe`],
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 50,
  },
  {
    id: 7,
    title: `Johnny English`,
    previewImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    genre: `Sci-Fi`,
    releaseDate: 2004,
    posterImage: `the-grand-budapest-hotel-poster.jpg`,
    backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
    ratingScore: 8.7,
    ratingCount: 2440,
    description: `Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.`,
    director: `director8`,
    starring: [`Bill Murray`, `Edward Norton`, `actor`, `actor2`],
    previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
    runTime: 40,
  },
];

const promoFilmMock = {
  id: 0,
  title: `The Grand Budapest Hotel`,
  previewImage: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  genre: `Drama`,
  releaseDate: 2014,
  posterImage: `the-grand-budapest-hotel-poster.jpg`,
  backgroundImage: `bg-the-grand-budapest-hotel.jpg`,
  ratingScore: 8.9,
  ratingCount: 240,
  description: `In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.`,
  director: `Wes Andreson`,
  starring: [`Bill Murray`, `Edward Norton`, `Jude Law`, `Willem Dafoe`],
  previewSrc: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`,
  runTime: 20,
  comments: [
    {
      id: 0,
      text: `Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.`,
      author: `Kate Muir`,
      date: `December 24, 2016`,
      rating: `8,9`
    },
    {
      id: 1,
      text: `Andersons films are too precious for some, but for those of us willing to
                lose ourselves in them, theyre a delight. The Grand Budapest Hotel is no different, except that he
                has added a hint of gravitas to the mix, improving the recipe.`,
      author: `Bill Goodykoontz`,
      date: `November 18, 2015`,
      rating: `8,0`
    }
  ],
};

jest.mock(`../user-block/user-block.jsx`, () => `user-block`);

it(`App render correct`, () => {
  const mockStore = configureStore([]);

  const store = mockStore({
    [Namespace.FILMS]: {
      movieCards: films,
      promoFilm: films[0],
      filterType: `All genres`,
      showingCardsCount: 8,
      isAppLoading: false
    },
    [Namespace.USER]: {
      authorizationStatus: false,
    },
    [Namespace.COMMENTS]: {
      comments: [],
      isCommentsLoaded: false
    }
  });

  const tree = renderer
    .create(
        <Provider store={store}>
          <App films={films} onCardClick={() => {}} promoFilm={promoFilmMock} login={() => {}} isLoading={false}/>
        </Provider>,
        {createNodeMock: () => ({})}
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
