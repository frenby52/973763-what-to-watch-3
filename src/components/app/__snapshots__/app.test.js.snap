// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App render correct 1`] = `
<Main
  films={
    Array [
      Object {
        "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
        "description": "In the 1930s, the Grand Budapest Hotel is a popular European ski resort, presided over by concierge Gustave H. (Ralph Fiennes). Zero, a junior lobby boy, becomes Gustave's friend and protege. Gustave prides himself on providing first-class service to the hotel's guests, including satisfying the sexual needs of the many elderly women who stay there. When one of Gustave's lovers dies mysteriously, Gustave finds himself the recipient of a priceless painting and the chief suspect in her murder.",
        "director": "Wes Andreson",
        "genre": "Drama",
        "id": 0,
        "posterImage": "the-grand-budapest-hotel-poster.jpg",
        "previewImage": "fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "ratingCount": 240,
        "ratingScore": 8.9,
        "releaseDate": 2014,
        "runTime": 20,
        "starring": Array [
          "Bill Murray",
          "Edward Norton",
          "Jude Law",
          "Willem Dafoe",
        ],
        "title": "The Grand Budapest Hotel",
      },
      Object {
        "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
        "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit.",
        "director": "director2",
        "genre": "Comedy",
        "id": 1,
        "posterImage": "the-grand-budapest-hotel-poster.jpg",
        "previewImage": "fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "ratingCount": 1240,
        "ratingScore": 8,
        "releaseDate": 2011,
        "runTime": 30,
        "starring": Array [
          "actor",
          "Edward Norton",
          "Jude Law",
          "Willem Dafoe",
        ],
        "title": "Bohemian Rhapsody",
      },
      Object {
        "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
        "description": "Cras aliquet varius magna, non porta ligula feugiat eget.",
        "director": "director3",
        "genre": "Horror",
        "id": 2,
        "posterImage": "the-grand-budapest-hotel-poster.jpg",
        "previewImage": "fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "ratingCount": 40,
        "ratingScore": 7,
        "releaseDate": 2010,
        "runTime": 50,
        "starring": Array [
          "Bill Murray",
          "actor",
          "Jude Law",
          "Willem Dafoe",
        ],
        "title": "Macbeth",
      },
      Object {
        "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
        "description": "Fusce tristique felis at fermentum pharetra.",
        "director": "director4",
        "genre": "Comedy",
        "id": 3,
        "posterImage": "the-grand-budapest-hotel-poster.jpg",
        "previewImage": "fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "ratingCount": 2240,
        "ratingScore": 1.9,
        "releaseDate": 2012,
        "runTime": 40,
        "starring": Array [
          "Bill Murray",
          "Edward Norton",
          "actor",
          "Willem Dafoe",
        ],
        "title": "Aviator",
      },
      Object {
        "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
        "description": "Aliquam id orci ut lectus varius viverra.",
        "director": "director5",
        "genre": "Crime",
        "id": 4,
        "posterImage": "the-grand-budapest-hotel-poster.jpg",
        "previewImage": "fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "ratingCount": 20,
        "ratingScore": 2.9,
        "releaseDate": 2013,
        "runTime": 20,
        "starring": Array [
          "Bill Murray",
          "Edward Norton",
          "Jude Law",
          "actor",
        ],
        "title": "We need to talk about Kevin",
      },
      Object {
        "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
        "description": "Nullam nunc ex, convallis sed finibus eget, sollicitudin eget ante.",
        "director": "director6",
        "genre": "Drama",
        "id": 5,
        "posterImage": "the-grand-budapest-hotel-poster.jpg",
        "previewImage": "fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "ratingCount": 240,
        "ratingScore": 5.9,
        "releaseDate": 2000,
        "runTime": 31,
        "starring": Array [
          "actor",
          "actor2",
          "Jude Law",
          "Willem Dafoe",
        ],
        "title": "What We Do in the Shadows",
      },
      Object {
        "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
        "description": "Phasellus eros mauris, condimentum sed nibh vitae, sodales efficitur ipsum.",
        "director": "director7",
        "genre": "Comedy",
        "id": 6,
        "posterImage": "the-grand-budapest-hotel-poster.jpg",
        "previewImage": "fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "ratingCount": 245,
        "ratingScore": 4.9,
        "releaseDate": 2001,
        "runTime": 50,
        "starring": Array [
          "Bill Murray",
          "actor",
          "actor2",
          "Willem Dafoe",
        ],
        "title": "Revenant",
      },
      Object {
        "backgroundImage": "bg-the-grand-budapest-hotel.jpg",
        "description": "Sed blandit, eros vel aliquam faucibus, purus ex euismod diam, eu luctus nunc ante ut dui.",
        "director": "director8",
        "genre": "Sci-Fi",
        "id": 7,
        "posterImage": "the-grand-budapest-hotel-poster.jpg",
        "previewImage": "fantastic-beasts-the-crimes-of-grindelwald.jpg",
        "previewSrc": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
        "ratingCount": 2440,
        "ratingScore": 8.7,
        "releaseDate": 2004,
        "runTime": 40,
        "starring": Array [
          "Bill Murray",
          "Edward Norton",
          "actor",
          "actor2",
        ],
        "title": "Johnny English",
      },
    ]
  }
  genre="Drama"
  onCardClick={[Function]}
  releaseDate={2014}
  title="The Grand Budapest Hotel"
/>
`;
